lib/document.js

```diff
@@ -2455,10 +2459,20 @@ Document.prototype.$__validate = function(pathsToValidate, options, callback) {
     pathDetails[0].filter((path) => this.isModified(path)) :
     pathDetails[0];
   const skipSchemaValidators = pathDetails[1];
-
   if (typeof pathsToValidate === 'string') {
     pathsToValidate = pathsToValidate.split(' ');
   }
+  if (pathsToSkip) {
+    // make a new variable that is paths intersection !options.pathsToSkip
+    let valPaths = new Set(paths);
```

Comment:
I'd recommend making `pathsToSkip` a set and [iterating over `paths` with a `for/of`](https://masteringjs.io/tutorials/fundamentals/array-iterate) instead of making `paths` a set and iterating over `pathsToSkip`. That's likely cleaner

Reply:
`paths = paths.filter(p => !pathsToSkip.has(p))`, see [filter arrays in JavaScript](https://masteringjs.io/tutorials/fundamentals/filter).

Reply:
Ok I see now but `_handlePathsToValidate` takes all the paths `paths` and then the paths to validate `valPaths` as the arguments. if I do that re-assignment if will mess up the functino execution.